### PR TITLE
Add BUILD_GRPC CMake flag and ROS-gRPC binary.

### PR DESCRIPTION
**Pull request by cschuet**

### CODE REVIEW

File: cartographer_ros/cartographer_ros/cartographer_grpc/node_grpc_main.cc [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+int main(int argc, char** argv) {
+  google::InitGoogleLogging(argv[0]);
+  google::ParseCommandLineFlags(&argc, &argv, true);
+
+  CHECK(!FLAGS_configuration_directory.empty())
+      << "-configuration_directory is missing.";
+  CHECK(!FLAGS_configuration_basename.empty())
+      << "-configuration_basename is missing.";
+
+  ::ros::init(argc, argv, "cartographer_grpc_node");
```
Comment (gaschler) [inline cartographer_ros/cartographer_ros/cartographer_grpc/node_grpc_main.cc]: Correct this name.

File: cartographer_ros/CMakeLists.txt [hunk 2]
```diff
@@ -38,6 +38,7 @@ set(PACKAGE_DEPENDENCIES
 find_package(cartographer REQUIRED)
 include("${CARTOGRAPHER_CMAKE_DIR}/functions.cmake")
 set(BUILD_SHARED_LIBS OFF)
+option(BUILD_GRPC "build Cartographer gRPC support" false)
```
[the remark this answers is not in the document]
Comment (cschuet) [reply cartographer_ros/CMakeLists.txt]: Done.

File: cartographer_ros/cartographer_ros/CMakeLists.txt [hunk 3]
```diff
@@ -91,3 +91,15 @@ install(TARGETS cartographer_pbstream_to_ros_map
   RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
 )
 
+if (${BUILD_GRPC})
```
[the remark this answers is not in the document]
Comment (cschuet) [reply cartographer_ros/cartographer_ros/CMakeLists.txt]: Done.

File: cartographer_ros/cartographer_ros/cartographer_grpc/node_grpc_main.cc (hunk 1 above)
Comment (cschuet) [reply cartographer_ros/cartographer_ros/cartographer_grpc/node_grpc_main.cc]: Done.

File: cartographer_ros/cartographer_ros/cartographer_grpc/node_grpc_main.cc [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  ::ros::start();
+
+  cartographer_ros::ScopedRosLogSink ros_log_sink;
+  cartographer_ros::cartographer_grpc::Run();
+  ::ros::shutdown();
```
[the remark this answers is not in the document]
Comment (cschuet) [reply cartographer_ros/cartographer_ros/cartographer_grpc/node_grpc_main.cc]: Done.

**Comment by cschuet** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user cschuet. Merge queue now has a length of 1.

**Comment by wally-the-cartographer** [comment]
Refusing to merge, since the combined checks state is failure

**Comment by cschuet** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user cschuet. Merge queue now has a length of 1.

**Pull request merged**